mongodb/driver.go

```diff
@@ -26,22 +25,76 @@ const (
 	int64Size                            = 64
 )
 
+// TLSConnectionConfig supplies the options for setting up a TLS based connection to the Mongo DB server
+// If the Mongo server certificate is to be validated (a major security breach not doing so), the VerifyCert
+// should be true, and the chain of CA certificates for the validation must be supplied -  in a file specified
+// by the absolute path given in the CACertChain attribute.
+// If the connection to the server is being made with an IP address, or via an SSH proxy
+// (such as with `dp ssh develop publishing 1 -p local-port:remote-host:remote-port`)
+// the real hostname should be supplied in the RealHostnameForSSH attribute. The real hostname is the
+// name of the server as attested by the server's x509 certificate. So in the above example of a connection via ssh
+// this would be the value of `remotehost`
+type TLSConnectionConfig struct {
```

Comment:
Hi David, I was looking at this as more of a convenience for clients of the library. They are free to embed the struct directly into their config, but they are also free to do as they do now, which is to copy from their ‘own managed’ config into the MongoConnectionConfig struct when initialising the mongo driver. I added because from what I can see most services will have the exact config attributes from the required attributes in MongoConnectionConfig , and then simply copy from their config into the MongoConnectionConfig (as I say they can continue to do this but if they do no other processing they can simply embed the given MongoConnectionConfig). I may be missing something here but if not I may leave as is?